clang/lib/AST/ExprConstantMeta.cpp

```diff
@@ -4532,6 +4532,47 @@ bool define_aggregate(APValue &Result, ASTContext &C, MetaActions &Meta,
   if (!Definition)
     return true;
 
+  if (Definition->isUnion()) {
```

Comment:
decided to put this logic here instead of `SemaReflect` because this code is specific to function and doesn't require access to `Sema`